### PR TITLE
Implemented getMetadata

### PR DESCRIPTION
**Pull request by dghelm**
Implemented getMetadata by calling getSkylinkUrl, doing a HEAD request on this, and then returning the JSON parse of the "skynet-file-metadata" header.

### CODE REVIEW

File: src/download.ts [hunk 1]
```diff
@@ -70,8 +70,17 @@ export function getHnsresUrl(domain: string, customOptions = {}) {
 
 export async function getMetadata(skylink: string, customOptions = {}) {
   const opts = { ...defaultDownloadOptions, ...this.customOptions, ...customOptions };
+  const url = this.getSkylinkUrl(skylink, opts);
+
+  console.log(url);
```
Comment (interstella-5555) [inline src/download.ts]: please remove `console.log` once this gets ready for merge

**Review by interstella-5555 — CHANGES_REQUESTED** [review]
That's a great contribution! Please add a test case for this function though, we need to make sure it works.

**Review by mrcnski — CHANGES_REQUESTED** [review]
Thanks, looks great! Just needs a quick test.

**Comment by dghelm** [comment]
Test added. Console Log removed.

Function modified to return an empty object if no "skynet-file-metadata" present. Should this instead throw an error?

File: src/download.test.ts [hunk 2]
```diff
@@ -118,6 +153,7 @@ describe("openFileHns", () => {
 
   beforeEach(() => {
     mock = new MockAdapter(axios);
+    const data = {};
```
Comment (mrcnski) [inline src/download.test.ts]: Superfluous line?

File: src/download.ts [hunk 3]
```diff
@@ -72,15 +72,22 @@ export async function getMetadata(skylink: string, customOptions = {}) {
...
-  console.log(url);
-
-  const response = await this.executeRequest({
-    ...opts,
-    method: "head",
-    url,
-  });
-
-  return JSON.parse(response.headers["skynet-file-metadata"]);
+  let skynetFileMetadata = {};
```
Comment (mrcnski) [inline src/download.ts]: Minor point: you can save a couple lines by returning `skynetFileMetadata` from within the `try` block. No need to declare it here.

**Review by mrcnski — APPROVED** [review]
Awesome, looks great! I ran a manual test and confirmed that your code works 😁

File: src/download.test.ts (hunk 2 above)
Comment (dghelm) [reply src/download.test.ts]: Yup! Will remove.

File: src/download.ts (hunk 3 above)
Comment (dghelm) [reply src/download.ts]: Good catch.

**Review by mrcnski — APPROVED** [review]

**Review by interstella-5555 — APPROVED** [review]
Looks great!

**Pull request merged**